cromwell-backend/src/main/scala/cromwell/backend/model/TaskDescriptor.scala

```diff
@@ -0,0 +1,24 @@
+package cromwell.backend.model
+
+import wdl4s.TaskOutput
+import wdl4s.command.CommandPart
+import wdl4s.values.WdlValue
+
+/**
+  * Represents basic data needed to execute a task in a defined backend.
+ *
+  * @param name Task name.
+  * @param user User who want to execute the task.
+  * @param commandTemplate Arguments for the execution.
+  * @param workingDir Task working directory.
+  * @param outputs Outputs generated by the execution.
+  * @param runtimeAttributes Configuration needed at runtime in order to execute a task.
+  */
+case class TaskDescriptor(name: String,
+                          user: String,
+                          commandTemplate: Seq[CommandPart], //TODO: Should be Wdl Agnostic
```

Comment:
My thought was to have the language support be pluggable as well as my belief is that if this is done correctly that it'd be easy to add another language (e.g. galaxy). So when a workflow is submitted somehow (magic? directly told?) Cromwell knows which language bindings to use and converts.

The conversion of expressions turns out to not be as simple as it seems. We've had some thought experiments both going down the scala expression path as well as conversion to javascript and using an in-JVM javascript engine. Both are doable but both have their own pros and cons. Yet another possibility is somehow allowing for multiple types of expressions.

There's a lot more nuance I started typing up but like I said, this isn't really the time nor the place to discuss the CWL issue.